woocommerce/class-sv-wc-plugin.php

```diff
@@ -77,29 +77,70 @@
 
 
 	/**
-	 * Initialize the plugin
+	 * Initialize the plugin.
 	 *
-	 * Optional args:
-	 *
-	 * + `dependencies` - array string names of required PHP extensions
-	 * + `function_dependencies` - array string names of required PHP functions
-	 *
-	 * Child plugin classes may add their own optional arguments
+	 * Child plugin classes may add their own optional arguments.
 	 *
 	 * @since 2.0.0
 	 * @param string $id plugin id
 	 * @param string $version plugin version number
-	 * @param array $args optional plugin arguments
+	 * @param array $args {
+	 *     optional plugin arguments
+	 *
+	 *     @type string $text_domain the plugin textdomain, used to set up translations
+	 *     @type array  $dependencies {
+	 *          the plugin's PHP dependencies
+	 *
+	 *          $type array $extensions the required PHP extensions
+	 *          $type array $functions  the required PHP functions
+	 *          $type array $settings   the required PHP settings, as `$setting => $value`
+	 *                                  String values are treated with direct
+	 *                                  comparison, integers as minimums
+	 *     }
+	 * }
 	 */
 	public function __construct( $id, $version, $args = array() ) {
 
 		// required params
 		$this->id          = $id;
 		$this->version     = $version;
 
-		if ( isset( $args['dependencies'] ) )                $this->dependencies = $args['dependencies'];
+		$default_dependencies = array(
```

Comment:
Might be useful to move these setters into their own method, outside of the constructor, to make it easier to unit test.